docs/Stream-Protocol-Spec.md

```diff
@@ -0,0 +1,88 @@
+
+stream! protocol
+======
+
+### definition of stream
+a protocol that facilitates data transmission between two swarm nodes, specifically targeting sequencial data in the form of a sequence of chunks as defined by swarm. the protocol should cater for the following requirements: 
+- client should be able to request arbitrary ranges from the server
+- client can be assumed to have some of the data already and therefore can opt in to selectivally request chunks based on their hashes
```

Comment:
aren't those two requests actually the same requirement? if client can request arbitrary ranges from the server, then it basically means that the client can selectively request chunks?

Reply:
not really, because bin index ranges can be asked for, for example, without a roundtrip, and that constitutes a valid request for a range of chunks.
for more granular control you have the possibility of a roundtrip